### PR TITLE
[BUGFIX] `TupleAzureBlobStoreBackend` no longer gives warning when obfuscating connection string

### LINKED ISSUE #5058: SQLAlchemy warning when not using SQL
**Describe the bug**
I'm using Azure Blob Storage as a backend and getting the following warning in the new version of ge 0.15.4. Don't think this warning showed up in versions prior to 0.15.

`WARNING:great_expectations.data_context.util:Something went wrong when trying to use SQLAlchemy to obfuscate URL: Could not parse rfc1738 URL from string 'BlobEndpoint=xxxxxx'`

**Expected behavior**
No warning of this is type should show when TupleAzureBlobStoreBackend backend is used.

**Environment (please complete the following information):**
 - System: Azure
 - Great Expectations Version: 0.15.4

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
## Changes proposed in this pull request:
- Azure blobstore connection strings don't follow the SqlAlchemy connection string pattern and was breaking the `PasswordMasker` method of obfuscating passwords.
- This adds a `_obfuscate_azure_blobstore_connection_string()` that will use a regex to substitute the `AccountKey` which is the secret portion of the connection string. ( Code was adopted from a snippet by user [@dvanbrug](https://github.com/dvanbrug) in the Issue discussion)
- Adds tests
- closes #5058 
### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/data_context/util.py [hunk 1]
```diff
@@ -198,6 +203,21 @@ def mask_db_url(cls, url: str, use_urlparse: bool = False, **kwargs) -> str:
...
+        )
+        matched: re.Match[str] | None = azure_conn_str_re.match(url)
+        try:
+            res = f"DefaultEndpointsProtocol={matched.group(2)};AccountName={matched.group(4)};AccountKey=***;EndpointSuffix={matched.group(8)}"
+            return res
+        except Exception as e:
+            logger.warning(
+                f"Something went wrong when trying to obfuscate URL for Azure: {e}"
+            )
+
```
Comment (anthonyburdi) [inline great_expectations/data_context/util.py]: I think here we should just raise the error instead of catching in a try except.

**Review by anthonyburdi — CHANGES_REQUESTED** [review]
Looks great, just a request to remove the try/except

File: great_expectations/data_context/util.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/data_context/util.py]: I agree with you on this one. I was trying to preserve the existing pattern, but think that this is our chance to "make things better". Update coming up

**Review by anthonyburdi — APPROVED** [review]
LGTM!

**Pull request merged**